### DISCUSSION TITLE
Is there a boilerplate for Auth + DB (so users get stored into db)

### DISCUSSION THREAD

**Discussion by dhuber666** [vercel/next.js#13432] [Help] [answered]
Hi guys,

I have a new web app I want to bring online quickly. It's not too complicated but it needs to handles auth and user related data. This time I want to try next.js

Is there a boilerplate that gets me up quickly? I don't care about the type of db. I just want to deploy it to vercel when finished and it should work :D

How are you creating this workflow these days with next.js?

Thx so much!

**Comment by brianlovin** [comment, marked as answer]
Yep! Check out the examples directory: https://github.com/vercel/next.js/tree/canary/examples

A bunch of options:
- https://github.com/vercel/next.js/tree/canary/examples/auth0
- https://github.com/vercel/next.js/tree/canary/examples/with-cookie-auth
- https://github.com/vercel/next.js/tree/canary/examples/with-cookie-auth-fauna
- https://github.com/vercel/next.js/tree/canary/examples/with-firebase-authentication
- https://github.com/vercel/next.js/tree/canary/examples/with-graphql-faunadb
- https://github.com/vercel/next.js/tree/canary/examples/with-firebase-authentication-serverless
- https://github.com/vercel/next.js/tree/canary/examples/with-iron-session
- https://github.com/vercel/next.js/tree/canary/examples/with-passport-and-next-connect

**Comment by hoangvvo** [comment]
I wrote a simple boilerplate at https://github.com/hoangvvo/nextjs-mongodb-app (deployable to Vercel in one-click using a button).

You can find several others at https://github.com/unicodeveloper/awesome-nextjs#boilerplates

**Reply by dhuber666** [reply]
Looks very nice. Thx

**Reply by dhuber666** [reply]
**EDIT** Nvm, I had to whiteliste my ip address on mongo atlas :D

Hi @hoangvvo  I followed your tutorial and also your github repo "nextjs-mongodb-app". I just want to use the mongodb part and so I did it the same as you. But when I go on a route that fetches something, it loads very long and I see this things in the command line:

```
the options [servers] is not supported
the options [caseTranslate] is not supported
the options [dbName] is not supported
the options [srvHost] is not supported
the options [credentials] is not supported
```

Then it breaks completely. According to SO and other posts this is when we open new connections constantly but that's why we have the check if it's already open in place right? But somehow it doesn't work. 

Maybe you can find the problem? I can't get it to work. 

This is my repo: https://github.com/dhuber666/copy-pasta/tree/mongo

I have everything like you in middleware/database and middleware/middleware

Where I actually use the db is in pages/api/people.ts

I would really appreciate your help!
